scrapy/addons.py

```diff
@@ -52,3 +52,12 @@ def load_settings(self, settings: Settings) -> None:
             },
             extra={"crawler": self.crawler},
         )
+
+    @classmethod
+    def load_early_settings(cls, settings: BaseSettings):
+        for clspath in build_component_list(settings["ADDONS"]):
+            try:
+                addoncls = load_object(clspath)
+                addoncls.update_spider_modules(settings)
+            except AttributeError:
+                pass
```

Comment:
Why this exception?

I’m thinking we should not capture any exception here, in line with the regular method, which only captures `NotConfigured` (which we should probably not capture here, it would be weird for users to have to raise that exception from 2 places instead of simply doing nothing in the class method).